hamocc/mo_bgcmean.F90

```diff
@@ -5,2354 +5,2216 @@
+    !
+    ! Arguments
+    integer, intent(in) :: pos(nbgcmax) ! position in 3d layer buffer
+    real,    intent(in) :: fld(idm,jdm) ! input data used for accumulation
+    !
+    ! Local varaibles
+    integer :: i,j,l,o
+    !
+    ! --- Check whether field should be accumulated
+    do o=1,nbgc
+      if (pos(o).EQ.0) cycle
+      !
+      !$OMP PARALLEL DO PRIVATE(l,i)
+      do j=1,jj
+        do l=1,isp(j)
+          do i=max(1,ifp(j,l)),min(ii,ilp(j,l))
+            bgct_bur(i,j,pos(o))=bgct_bur(i,j,pos(o))+fld(i,j)
+          enddo
+        enddo
+      enddo
+      !$OMP END PARALLEL DO
+    enddo
+    !
+  end subroutine accbur
+
+
+  subroutine finsrf(posacc,poswgt)
+    !
+    ! --- ------------------------------------------------------------------
+    ! --- Description: finalise accumulation of weighted 2d fields
+    ! --- ------------------------------------------------------------------
+    !
+    ! Arguments
+    integer, intent(in)  :: posacc ! position of accumulated field in buffer
+    integer, intent(in)  :: poswgt ! position of accumulated weights
+    !
+    ! Local variables
+    integer :: i,j,l
+    real, parameter :: epsil=1e-11
+    !
+    ! --- Check whether field should be initialised
+    if (posacc.EQ.0) RETURN
+    !
+    !$OMP PARALLEL DO PRIVATE(l,i)
+    do j=1,jj
+      do l=1,isp(j)
+        do i=max(1,ifp(j,l)),min(ii,ilp(j,l))
+          bgcm2d(i,j,posacc)=bgcm2d(i,j,posacc)/max(epsil,bgcm2d(i,j,poswgt))
+        enddo
+      enddo
+    enddo
+    !$OMP END PARALLEL DO
+    !
+  end subroutine finsrf
+
+
+
+  subroutine finlyr(posacc,poswgt)
+    !
+    ! --- ------------------------------------------------------------------
+    ! --- Description: finalise accumulation of weighted 3d layer fields
+    ! --- ------------------------------------------------------------------
+    !
+    ! Arguments
+    integer, intent(in)  :: posacc ! position of accumulated field in buffer
+    integer, intent(in)  :: poswgt ! position of accumulated weights
+    !
+    ! Local variables
+    integer :: i,j,k,l
+    real, parameter :: epsil=1e-11
+    !
+    ! --- Check whether field should be initialised
+    if (posacc.EQ.0) RETURN
+    !
+    do k=1,kk
+      !$OMP PARALLEL DO PRIVATE(l,i)
+      do j=1,jj
+        do l=1,isp(j)
+          do i=max(1,ifp(j,l)),min(ii,ilp(j,l))
+            if (bgcm3d(i,j,k,poswgt).GT.epsil) THEN
+              bgcm3d(i,j,k,posacc)=bgcm3d(i,j,k,posacc)/bgcm3d(i,j,k,poswgt)
+            else
+              bgcm3d(i,j,k,posacc)=nf90_fill_double
+            endif
+          enddo
+        enddo
+      enddo
+      !$OMP END PARALLEL DO
+    enddo
+    !
+  end subroutine finlyr
+
+
+
+  subroutine wrtsrf(pos,frmt,sfac,offs,cmpflg,vnm)
+    !
+    ! --- ------------------------------------------------------------------
+    ! --- Description: writes diagnostic 2d field to file
+    ! --- ------------------------------------------------------------------
+    !
+    ! Arguments
```

Comment:
Pleasew update the comments according to the former comments:
```
! --- Arguments:
! ---   int  pos      (in)     : variable position in common buffer
! ---   int  frmt     (in)     : format/precision of output 
! ---                            0=field is not written  
! ---                            2=field is written as int2 with scale 
! ---                              factor and offset 
! ---                            4=field is written as real4
! ---                            8=field is written as real8
! ---   real sfac     (in)     : user def.NE. scale factor to be applied   
! ---   real offs     (in)     : user def.NE. offset to be added 
! ---   int  cmpflg   (in)     : compression flag; only wet points are 
! ---                            written IF flag is set to 1 
! ---   char vnm      (in)     : variable name used in nc-file 
```

Reply:
Done.